### PR TITLE
Add Time, Status Indexes to JobResult

### PR DESCRIPTION
**Pull request by bryanculver**
# Closes: #DNE
# What's Changed

- Added indexes to `JobResult` to speed up list and homepage performance.

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [NA] Attached Screenshots, Payload Example
- [NA] Unit, Integration Tests
- [NA] Documentation Updates (when adding/changing features)
- [NA] Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/models/jobs.py [hunk 1]
```diff
@@ -586,6 +586,32 @@ def __init__(self, *args, **kwargs):
...
         get_latest_by = "created"
+        indexes = [
+            models.Index(
+                name="extras_jobresult_created_idx",
+                fields=["created"],
+            ),
+            models.Index(
+                name="extras_jobresult_rcreated_idx",
+                fields=["-created"],
+            ),
```
Comment (glennmatthews) [inline nautobot/extras/models/jobs.py:597]: Ignorant question - is there actually a need/value to indexing the same field in both directions like this? I would have naively assumed that one index would cover both directions. (and in that case, wouldn't adding `db_index=True` on the `created` field solve this more succinctly?)

Comment (jathanism) [reply nautobot/extras/models/jobs.py:597]: I would rather see the forward indexes done on the field using `db_index=True`, only leaving the reverse indexes here in the `Meta.indexes`.

Comment (bryanculver) [reply nautobot/extras/models/jobs.py]: Sorry it took me a while to update. @glennmatthews @jathanism updated from feedback.

**Review by glennmatthews — APPROVED** [review]
Needs blackening, otherwise looks good. Thanks!

**Comment by bryanculver** [comment]
> Needs blackening, otherwise looks good. Thanks!

🤦

**Pull request merged**